api/src/pcapi/connectors/api_adresse.py

```diff
@@ -45,6 +49,74 @@ class AddressInfo(pydantic_v1.BaseModel):
     score: float
 
 
+class ResultColumn(enum.Enum):
+    LATITUDE = "latitude"
+    LONGITUDE = "longitude"
+    RESULT_CITY = "result_city"
+    RESULT_CITYCODE = "result_citycode"
+    RESULT_CONTEXT = "result_context"
+    RESULT_DISTRICT = "result_district"
+    RESULT_HOUSENUMBER = "result_housenumber"
+    RESULT_ID = "result_id"
+    RESULT_LABEL = "result_label"
+    RESULT_NAME = "result_name"
+    RESULT_OLDCITY = "result_oldcity"
+    RESULT_OLDCITYCODE = "result_oldcitycode"
+    RESULT_POSTCODE = "result_postcode"
+    RESULT_SCORE = "result_score"
+    RESULT_SCORE_NEXT = "result_score_next"
+    RESULT_STATUS = "result_status"
+    RESULT_STREET = "result_street"
+    RESULT_TYPE = "result_type"
+
+
+def format_q(address: str, city: str) -> str:
+    """
+    This method formats the search address (q) in order to get the most accurate results from BAN.
+
+    Hence, this method:
+     - only uses address and city, without postcode.
+       In several cases, including postcode can induce errors in results.
+       Some cities have several postcodes and results may differ depending on the given postcode.
+       Examples (q --> result):
+       With postcode: '33 Boulevard Clemenceau 38000 Grenoble' --> '33 Boulevard Gambetta 38000 Grenoble'
+       Without postcode: '33 Boulevard Clemenceau Grenoble' --> '33 Boulevard Clemenceau 38100 Grenoble'
+
+     - normalizes / formats address and city to get as close as possible to the address format used by BAN.
+       It also makes sure neither address nor city contains a comma, as it would break the CSV payload.
+       Examples (q --> result):
+       Not normalized: '105 RUE DES HAIES PARIS 20' --> 'Voie Ao/20 75020 Paris'
+       Normalized: '105 Rue Des Haies Paris' --> '105 Rue des Haies 75020 Paris'
+    """
+    # Address:
+    address = address.strip()
+    address = address.replace(",", "")
```

Comment:
Tu sais si on a des adresses sans espace après la virgule ex: 23,rue de la boétie,Paris,92300, après le replace ce sera 23rue de la boétieParis92300

Reply:
Bien vu ! J'ai regardé, il se trouve qu'il n'y a pas d'adresse dans ce cas là, mais c'est peut-être plus propre de faire un :
```python
address = address.replace(",", " ")
address = ' '.join(address.split())  # substitute multiple whitespace with a single one
```

Reply:
C'est parfait
`In [1]: s = '23,rue collange'

In [2]: s.replace(',', " ")
Out[2]: '23 rue collange'

In [3]: s.replace(',', " ").split()
Out[3]: ['23', 'rue', 'collange']

In [4]: 

In [4]: s = '23, rue collange'

In [5]: s.replace(',', " ").split()
Out[5]: ['23', 'rue', 'collange']
`